### PR TITLE
fix gh-1224 add .prev and :set_prev to box.error

### PR DESCRIPTION
**Pull request by Onvember**

### CODE REVIEW

File: doc/reference/reference_lua/box_error.rst [hunk 1]
```diff
@@ -174,5 +180,62 @@ Below is a list of all ``box.error`` functions.
             line: 1
         ...
 
+.. _box_error-error_object:
+
+.. class:: error_object
+
+    Errors can be organized into lists. To achieve this Lua table representing
```
Comment (lenkis) [inline doc/reference/reference_lua/box_error.rst:187]:
```suggestion
    Errors can be organized into lists. To achieve this, a Lua table representing an
```

File: doc/reference/reference_lua/box_error.rst [hunk 2]
```diff
@@ -174,5 +180,62 @@ Below is a list of all ``box.error`` functions.
...
+    error object has ``.prev`` field and ``e:set_prev(err)`` method.
+
+    .. _box_error-prev:
+
+    .. data:: prev
+
+        Return a previous error if any exist.
```
Comment (lenkis) [inline doc/reference/reference_lua/box_error.rst:194]:
```suggestion
        Return a previous error, if any.
```

File: doc/reference/reference_lua/box_error.rst [hunk 3]
```diff
@@ -174,5 +180,62 @@ Below is a list of all ``box.error`` functions.
...
+
+    .. _box_error-set_prev:
+
+    .. method:: set_prev(error object)
+
+        Set an error as previous error. Accepts an `error object` or `nil`.
```
Comment (lenkis) [inline doc/reference/reference_lua/box_error.rst:200]:
```suggestion
        Set an error as the previous error. Accepts an ``error object`` or ``nil``.
```

File: doc/reference/reference_lua/box_error.rst [hunk 4]
```diff
@@ -174,5 +180,62 @@ Below is a list of all ``box.error`` functions.
...
+    Cycles are not allowed for error lists:
+
+    .. code-block:: tarantoolsession
+
+        tarantool> e2:set_prev(e1)
+        ---
+        - error: 'builtin/error.lua:147: Cycles are not allowed'
+        ...
+
+    Setting previous error does not erase its own previous members:
```
Comment (lenkis) [inline doc/reference/reference_lua/box_error.rst:229]:
```suggestion
    Setting the previous error does not erase its own previous members:
```

File: doc/reference/reference_lua/box_error.rst [hunk 5]
```diff
@@ -174,5 +180,62 @@ Below is a list of all ``box.error`` functions.
...
+    .. code-block:: Lua
 
+        -- e1 -> e2 -> e3 -> e4
+        e1:set_prev(e2)
+        e2:set_prev(e3)
+        e3:set_prev(e4)
+        e2:set_prev(e5)
+        -- Now there are two lists: e1->e2->e5 and e3->e4
 
+    Stacked diagnostics is also supported by IProto protocol. See details
```
Comment (lenkis) [inline doc/reference/reference_lua/box_error.rst:240]:
```suggestion
    The iProto protocol also supports stacked diagnostics. See details
```

File: doc/reference/reference_lua/box_error.rst [hunk 6]
```diff
@@ -40,6 +40,12 @@ Below is a list of all ``box.error`` functions.
     | :ref:`box.error.new()                | Create an error but do not      |
     | <box_error-new>`                     | throw                           |
     +--------------------------------------+---------------------------------+
+    | :ref:`error_object.prev              | Return a previous error         |
```
Comment (lenkis) [inline doc/reference/reference_lua/box_error.rst:43]:
```suggestion
    | :ref:`error_object.prev              | Return the previous error       |
```

File: doc/reference/reference_lua/box_error.rst [hunk 7]
```diff
@@ -40,6 +40,12 @@ Below is a list of all ``box.error`` functions.
...
+    | <box_error-prev>`                    |                                 |
+    +--------------------------------------+---------------------------------+
+    | :ref:`error_object.set_prev()        | Set an error as previous error  |
```
Comment (lenkis) [inline doc/reference/reference_lua/box_error.rst:46]:
```suggestion
    | :ref:`error_object.set_prev()        | Set the previous error          |
```

File: doc/reference/reference_lua/box_error.rst (hunk 3 above)
Comment (lenkis) [inline doc/reference/reference_lua/box_error.rst:200]:
```suggestion
        Set an error as the previous error. Accepts an ``error object`` or ``nil``.
```

Comment (Onvember) [reply doc/reference/reference_lua/box_error.rst]: fixes in 931a40c

File: locale/ru/LC_MESSAGES/reference/reference_lua/box_error.po [hunk 8]
```diff
@@ -334,44 +347,71 @@ msgid ""
...
-#~ "tarantool> box.error.last()\n"
-#~ "---\n"
-#~ "- nil"
-#~ msgstr ""
+msgid ""
+"Errors can be organized into lists. To achieve this, a Lua table "
+"representing an error object has ``.prev`` field and ``e:set_prev(err)`` "
+"method."
+msgstr ""
+"Ошибки можно записывать в списки. Чтобы сделатьт это, у Lua-таблицы, "
```
Comment (lenkis) [inline locale/ru/LC_MESSAGES/reference/reference_lua/box_error.po:355]:
```suggestion
"Ошибки можно записывать в списки. Для этого у Lua-таблицы, "
```

File: locale/ru/LC_MESSAGES/reference/reference_lua/box_error.po [hunk 9]
```diff
@@ -334,44 +347,71 @@ msgid ""
...
+"---\n"
+"...\n"
+"tarantool> e1.prev\n"
+"---\n"
+"- cause of cause\n"
+"..."
+msgstr ""
+
+msgid "Cycles are not allowed for error lists:"
+msgstr "Циклы недопустимы для списков ошибок:"
```
Comment (lenkis) [inline locale/ru/LC_MESSAGES/reference/reference_lua/box_error.po:386]:
```suggestion
msgstr "Циклы для списков ошибок недопустимы:"
```

File: locale/ru/LC_MESSAGES/reference/reference_lua/box_error.po [hunk 10]
```diff
@@ -334,44 +347,71 @@ msgid ""
...
+msgid ""
+"tarantool> e2:set_prev(e1)\n"
+"---\n"
+"- error: 'builtin/error.lua:147: Cycles are not allowed'\n"
+"..."
+msgstr ""
+
+msgid "Setting the previous error does not erase its own previous members:"
+msgstr ""
+"Установка предыдущей ошибки не подтягивает за ней ее собственные "
```
Comment (lenkis) [inline locale/ru/LC_MESSAGES/reference/reference_lua/box_error.po:397]:
```suggestion
"Установка предыдущей ошибки подтягивает за ней ее собственные "
```

**Pull request merged**